providers/src/airflow/providers/amazon/aws/operators/ecs.py

```diff
@@ -661,7 +661,9 @@ def _aws_logs_enabled(self):
         return self.awslogs_group and self.awslogs_stream_prefix
 
     def _get_logs_stream_name(self) -> str:
-        return f"{self.awslogs_stream_prefix}/{self.container_name}/{self._get_ecs_task_id(self.arn)}"
+        if self.awslogs_stream_prefix and self.container_name and not self.awslogs_stream_prefix.endswith(self.container_name):
+                return f"{self.awslogs_stream_prefix}/{self.container_name}/{self._get_ecs_task_id(self.arn)}"
```

Comment:
`endswith` is too specific, I think we should check whether the container name is contained in the prefix

```suggestion
        if self.awslogs_stream_prefix and self.container_name and self.container_name not in self.awslogs_stream_prefix:
            return f"{self.awslogs_stream_prefix}/{self.container_name}/{self._get_ecs_task_id(self.arn)}"
```

Reply:
Per #43130 if log streams are automatically created like `prefix-name/container-name/ecs-task-id` then the content of `prefix-name` is only important to us if it contains `container-name` as a suffix (it should actually be `/container-name` now that I think about it).

If we just use `in` as our specifier then any prefix containing a container name would match, which would break our logic for determining the log stream name, e.g. `foobar_container_name_foobar` would become `foobar_container_name_foobar/container_name/task_id` by ECS but we would return `foobar_container_name_foobar/task_id` instead.

Reply:
Fair enough. I think you are right. Though, I think you have an indentation issue on the return level, please remove one

Reply:
Done, thanks.

Reply:
I was going to make the same suggestion, but your reasoning works for me.